api/jobs.go

```diff
@@ -933,6 +933,16 @@ type JobSubmission struct {
 	Variables string
 }
 
+type JobUIConfig struct {
+	Description string       `hcl:"description,optional"`
+	Links       []*JobUILink `hcl:"link,block"`
+}
```

Comment:
This likely needs a `Copy` and `Canonicalize` method, which is called from the same method on `Job`.